backend/infrahub/message_bus/operations/event/branch.py

```diff
@@ -41,6 +45,19 @@ async def merge(message: messages.EventBranchMerge, service: InfrahubServices) -
         messages.TriggerArtifactDefinitionGenerate(branch=message.target_branch),
         messages.TriggerGeneratorDefinitionRun(branch=message.target_branch),
     ]
+    component_registry = get_component_registry()
+    default_branch = registry.get_branch_from_registry()
+    diff_repository = await component_registry.get_component(DiffRepository, db=service.database, branch=default_branch)
+    # send diff update requests for every branch-tracking diff
+    branch_diff_roots = await diff_repository.get_empty_roots(base_branch_names=[message.target_branch])
```

Comment:
at this point, we can only merge a branch into `main` (or whatever the default is named). when a merge happens, we should update every diff that is tracking a branch so that they pull in whatever changes were added to `main`. this logic would need to change if we allowed a deeper branch hierarchy

Reply:
How expensive in terms of time are these operations? I'm thinking mainly that these event type of messages are supposed to be completed very fast and I'm wondering if we should instead trigger these calls to happen in another message. It doesn't have to be in this PR or release but could be nice to circle back to later.

Another potential issue here is that the `TriggerGeneratorDefinitionRun` message could lead to additional changes in the target branch. So after the async tasks have been reworked I think we want to consider the workflow here so that we request the Diff update to happen after any generators have completed.

Reply:
fortunately, this is a pretty cheap operation, but you're concern about order of execution in the pipeline is definitely something we need to take into account... somewhere